cognitive/src/test/scala/com/microsoft/azure/synapse/ml/cognitive/split2/SpeakerEmotionInferenceSuite.scala

```diff
@@ -0,0 +1,99 @@
+        "</mstts:express-as> she said.\"</voice></speak>\n")
+  )
+
+  lazy val df: DataFrame = testData.map(e => e._1).toSeq.toDF("Input")
+
+  test("basic") {
+    ssmlGenerator.transform(df).collect().map(row => {
+      assert(testData.get(row.getString(0)).getOrElse("").equals(row.getString(2)))
+    })
+  }
+
+  val ssmlFormatTestData = Map[(String, String), String](
+    ((""""A", "B", "C"""",
+      """{"IsValid":true,"Conversations":[
+        {"Content":"\"A\"","Begin":0,"End":3,"Role":"male","Style":"calm"},
+        {"Content":"\"B\"","Begin":5,"End":8,"Role":"male","Style":"calm"},
+        {"Content":"\"C\"","Begin":10,"End":13,"Role":"male","Style":"calm"}]}""") ->
+      ("""<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' """ +
+        """xmlns:mstts='https://www.w3.org/2001/mstts' xml:lang='en-US'><voice name='en-US-JennyNeural'>""" +
+        """<mstts:express-as role='male' style='calm'>"A"</mstts:express-as>, """ +
+        """<mstts:express-as role='male' style='calm'>"B"</mstts:express-as>, """ +
+        """<mstts:express-as role='male' style='calm'>"C"</mstts:express-as></voice></speak>""" + "\n")),
+    (("""Z"A"Z"B"Z"C"Z""",
+      """{"IsValid":true,"Conversations":[
+        {"Content":"\"A\"","Begin":1,"End":4,"Role":"male","Style":"calm"},
+        {"Content":"\"B\"","Begin":5,"End":8,"Role":"male","Style":"calm"},
+        {"Content":"\"C\"","Begin":9,"End":12,"Role":"male","Style":"calm"}]}""") ->
+      ("""<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' """ +
+        """xmlns:mstts='https://www.w3.org/2001/mstts' xml:lang='en-US'><voice name='en-US-JennyNeural'>Z""" +
+        """<mstts:express-as role='male' style='calm'>"A"</mstts:express-as>Z<mstts:express-as role='male' """ +
+        """style='calm'>"B"</mstts:express-as>Z<mstts:express-as role='male' style='calm'>"C"""" +
+        """</mstts:express-as>Z</voice></speak>""" + "\n")),
+    ((""""A""B""C"""",
+      """{"IsValid":true,"Conversations":[
+        {"Content":"\"A\"","Begin":0,"End":3,"Role":"male","Style":"calm"},
+        {"Content":"\"B\"","Begin":3,"End":6,"Role":"male","Style":"calm"},
+        {"Content":"\"C\"","Begin":6,"End":9,"Role":"male","Style":"calm"}]}""") ->
+      ("""<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' """ +
+        """xmlns:mstts='https://www.w3.org/2001/mstts' xml:lang='en-US'>""" +
+        """<voice name='en-US-JennyNeural'><mstts:express-as role='male' style='calm'>"A"""" +
+        """</mstts:express-as><mstts:express-as role='male' style='calm'>"B"</mstts:express-as>""" +
+        """<mstts:express-as role='male' style='calm'>"C"</mstts:express-as></voice></speak>""" + "\n")),
+  )
+
+  test("formatSSML") {
+    ssmlFormatTestData.map(test => {
+      val result = ssmlGenerator.formatSSML(
+        test._1._1,
+        "en-US",
+        "en-US-JennyNeural",
+        test._1._2)
+      assertResult(test._2)(result)
+    })
+  }
+
```

Comment:
maybe we should add a composition test here that shows that it can be directly pipes into TTS